File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        curr = ""
        stack = []
        
        for p in path + "/":
            if p == "/":
                if curr == "..":
                    if stack:
                        stack.pop()
                elif curr != "" and curr != ".":
                    stack.append(curr)
                curr = ""
            else:
                curr += p
        return  "/" + "/".join(stack)
        

File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    const stack = [];
    const pathItems = path.split("/");
    
    for (let item of pathItems) {
        if (!item || item === ".") {
            continue;
        } else if (item === "..") {
            stack.pop();
        } else {
            stack.push(item)
        }
    }
    
    return "/" + stack.join("/");
};